Assets/Scripts/PerkUIManager.cs

```diff
@@ -0,0 +1,106 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PerkUIManager : MonoBehaviour
+{
+    [SerializeField] GameObject perkPrefab;
+
+    IEnumerator enumerator = null;
+
+    GameObject selectedPerk = null;
+
+    public enum Type
+    {
+        ExtraSummon,
+        AddHealth,
+        AddSummonHealth
+    }
+    Type type;
+
+    public static PerkUIManager Instance
+    {
+        get;
+        private set;
+    }
+
+    private void Awake()
+    {
+        if (Instance != null && Instance != this)
+        {
+            Destroy(this);
+        }
+        else
+        {
+            Instance = this;
+        }
+    }
+
+    private void Start()
+    {
+        
+    }
+
+    private void Update()
+    {
+        if (enumerator == null)
+        {
+            enumerator = PerkSelection();
+            StartCoroutine(enumerator);
+        }
+    }
+
+    IEnumerator PerkSelection()
```

Comment:
create a level up event and hook this method to it and remove it from the update, also don't make it a coroutine